js/client-chat.js

```diff
@@ -746,28 +722,20 @@
 					allHighlights[Config.server.id] = serverHighlights;
 					Tools.prefs('highlights', allHighlights);
 				} else {
-					if (target === 'delete') {
-						serverHighlights['globalhls'] = [];
-						allHighlights[Config.server.id] = serverHighlights;
-						Tools.prefs('highlights', allHighlights);
-						this.add("All highlights cleared");
-					} else if (target === 'show' || target === 'list') {
-						// Shows a list of the current highlighting words
-						if (serverHighlights['globalhls'].length > 0) {
-							this.add("Current global highlight list: " + serverHighlights['globalhls'].join(", "));
-						} else {
-							this.add('Your global highlight list is empty.');
-						}
-					} else if (target === 'roomdelete') {
+					if (target === 'delete' || target === 'roomdelete') {
+						var thisroom = (target === 'delete' ? 'globalhls' : (this.type === 'battle' ? 'battlehls' : this.id));
+						var textending = (target === 'delete' ? 'All highlights cleared.' : (this.type === 'battle' ? 'All battle highlights cleared.' : 'All highlights cleared in ' + this.id + '.'));
 						serverHighlights[thisroom] = [];
 						allHighlights[Config.server.id] = serverHighlights;
 						Tools.prefs('highlights', allHighlights);
-						this.add("All room highlights cleared" + textending + ".");
-					} else if (target === 'roomshow' || target === 'roomlist') {
+						this.add(textending);
+					} else if (target === 'show' || target === 'list' || target === 'roomshow' || target === 'roomlist') {
+						var thisroom = (target === 'show' || target === 'list' ? 'globalhls' : (this.type === 'battle' ? 'battlehls' : this.id));
+						var textending = (target === 'show' || target === 'list' ? 'urrent global highlight list' : 'urrent highlight list for this room');
 						if (serverHighlights[thisroom].length > 0) {
-							this.add("Current room highlight list: " + serverHighlights[thisroom].join(", "));
+							this.add('C' + textending + ': ' + serverHighlights['globalhls'].join(", "));
 						} else {
-							this.add('Your highlight list in this room is empty.');
+							this.add('Your c' + textending + 'is empty.');
```

Comment:
... this is weird

if both lines of the options first word is "current", then why are you breaking it up like this..?